autoload/vundle/installer.vim

```diff
@@ -393,6 +393,7 @@ func! s:make_sync_command(bang, bundle) abort
                   \ 'git fetch',
                   \ 'git reset --hard origin/HEAD',
```

Comment:
`'git reset --hard origin/'.a:bundle.revision,`
I think this is what you want (which was done in line 396 in a wrong way) ?